src/odb/include/odb/db.h

```diff
@@ -7520,31 +7594,105 @@ class dbModInst : public dbObject
   static dbSet<dbModInst>::iterator destroy(dbSet<dbModInst>::iterator& itr);
 
   static dbModInst* getModInst(dbBlock* block_, uint dbid_);
-
-  std::string getName() const;
+  dbModITerm* getdbModITerm(dbId<dbModITerm>);
 
   std::string getHierarchicalName() const;
+
+  bool getPinAtIx(unsigned ix, dbModITerm*& ret) const;
+  bool findModITerm(const char* name, dbModITerm*& ret);
+
+  dbSet<dbModITerm> getModITerms();
+
   // User Code End dbModInst
 };
 
+class dbModITerm : public dbObject
+{
+ public:
+  void setFlags(uint flags);
+
+  uint getFlags() const;
+
+  dbModInst* getParent() const;
+
+  void setNet(dbModNet* net);
+
+  dbModNet* getNet() const;
+
+  void setNextNetModiterm(dbModITerm* next_net_moditerm);
+
+  dbModITerm* getNextNetModiterm() const;
+
+  void setPrevNetModiterm(dbModITerm* prev_net_moditerm);
+
+  dbModITerm* getPrevNetModiterm() const;
+
+  void setNextEntry(dbModITerm* next_entry);
+
+  dbModITerm* getNextEntry() const;
+
+  // User Code Begin dbModITerm
+  void setSigType(dbSigType type);
+  dbSigType getSigType();
+  void setIoType(dbIoType type);
+  dbIoType getIoType();
+
+  static dbModITerm* create(dbModInst* parentInstance, const char* name);
+  bool connect(dbModNet*);
+  char* getName();
+  // User Code End dbModITerm
+};
+
+class dbModNet : public dbObject
+{
+ public:
+  dbModule* getParent() const;
+
+  // User Code Begin dbModNet
+  dbSet<dbModITerm> getModITerms();
+  dbSet<dbModBTerm> getModBTerms();
+  dbSet<dbITerm> getITerms();
+  dbSet<dbBTerm> getBTerms();
+
+  static dbModNet* create(dbModule* parentModule, const char* name);
+  const char* getName() const;
+  dbModBTerm* connectedToModBTerm();
```

Comment:
what is this for?

Reply:
dbModNet create -- needed to create a hierarchical net in a module.
getName -- gets the name of the net
connectedToModBTerm -- used for telling if a modNet is hooked to a port on a module (a bterm) for hierarchical traversal.